frontend/catalyst/qfunc.py

```diff
@@ -62,6 +63,11 @@
 logger.addHandler(logging.NullHandler())
 
 
+def _get_device_shots(dev):
```

Comment:
Can we use this [here](https://github.com/PennyLaneAI/catalyst/blob/mcm-kwargs/frontend/catalyst/jax_tracer.py#L661) and [here](https://github.com/PennyLaneAI/catalyst/blob/mcm-kwargs/frontend/catalyst/device/qjit_device.py#L149)?